### PR TITLE
feat: `IsNewKurtosisVersionAvailable` and `UpgradeKurtosisVersion` GRPC endpoints added

### PR DESCRIPTION
**Pull request by leoporoli**
## Description
Adding this new endpoint because it will needed in the cloud-backend to implement the endpoints

## Is this change user facing?
NO

## References (if applicable)
This is part of the `upgrade Kurtosis from UI` project

### CODE REVIEW

File: cloud/api/protobuf/kurtosis_backend_server_api.proto [hunk 1]
```diff
@@ -281,4 +283,17 @@ message  AddAliasRequest {
   string access_token=1;
   Port port=2;
   string alias=3;
-}
\ No newline at end of file
+}
+
+message IsNewKurtosisVersionAvailableRequest {
+  string current_kurtosis_version = 2;
```
Comment (laurentluce) [inline cloud/api/protobuf/kurtosis_backend_server_api.proto:290]: Those requests go through the cloud gateway afaik so it should contain an access token.

File: cloud/api/protobuf/kurtosis_backend_server_api.proto [hunk 2]
```diff
@@ -281,4 +283,17 @@ message  AddAliasRequest {
...
+}
+
+message IsNewKurtosisVersionAvailableResponse {
+  bool isAvailable = 1;
+  string latestVersion = 2;
+}
+
+message UpgradeKurtosisVersionRequest {
+  string api_key = 1;
```
Comment (laurentluce) [inline cloud/api/protobuf/kurtosis_backend_server_api.proto]: If this is coming from the EM UI then it should take an access token (jwt).

File: cloud/api/protobuf/kurtosis_backend_server_api.proto (hunk 1 above)
Comment (leoporoli) [reply cloud/api/protobuf/kurtosis_backend_server_api.proto:290]: @laurentluce access token added

File: cloud/api/protobuf/kurtosis_backend_server_api.proto (hunk 2 above)
Comment (leoporoli) [reply cloud/api/protobuf/kurtosis_backend_server_api.proto]: @laurentluce access token added

**Review by laurentluce — APPROVED** [review]

**Pull request merged**